packages/gatsby-remark-embed-snippet/README.md

```diff
@@ -1,40 +1,115 @@
 # gatsby-remark-embed-snippet
 
-Embeds the contents of specified files within Prism-formatted HTML blocks.
+Embeds the contents of specified files as code-snippets.
 
-## Overview
+## Installation
+
+**Note**: This plugin depends on [gatsby-remark-prismjs](https://www.gatsbyjs.org/packages/gatsby-remark-prismjs/) and [gatsby-transformer-remark](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/) plugins
+
+```shell
+# npm
+npm install --save gatsby-remark-embed-snippet gatsby-remark-prismjs gatsby-transformer-remark
+
+# yarn
+yarn add gatsby-remark-embed-snippet gatsby-remark-prismjs gatsby-transformer-remark
+```
+
+## Configuration
+
+> **Important**: _You must add `gatsby-remark-embed-snippet` before `gatsby-remark-prismjs` in your plugins array!_
+> Otherwise, this plugin will not work because the code-snippet files first need to be inlined before it can be transformed into code blocks.
+> For more information, see [gatsby-remark-prismjs](https://www.gatsbyjs.org/packages/gatsby-remark-prismjs/).
+
+To use `gatsby-remark-embed-snippet` plugin:
+
+```js
+// gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-transformer-remark`,
+      options: {
+        plugins: [
+          {
+            resolve: `gatsby-remark-embed-snippet`,
+            options: {},
+          },
+          {
+            resolve: `gatsby-remark-prismjs`,
+            options: {},
+          },
+        ],
+      },
+    },
+  ],
+}
+```
+
+## Plugin option
+
+| option      | description                                                                                                                                                                             |
+| ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `directory` | Specify the directory where the code snippet files are located. If this option is omitted, this plugin will look for the code snippet files in the same directory as the markdown file. |
+
+### Example 1: Specify that code snippet files are under the root directory
 
-### Embedding Files
+```js
+// gatsby-config.js
+{
+  resolve: `gatsby-remark-embed-snippet`,
+  options: {
+    directory: `.`
+  }
+},
+```
 
-For example, given the following project directory structure:
+### Example 2: Specify that code snippet files are under a directory called `snippets`
 
+```js
+// gatsby-config.js
+{
+  resolve: `gatsby-remark-embed-snippet`,
+  options: {
+    directory: `./snippets/`
```

Comment:
The previous example and other config have always the `${__dirname}` in path. There are any changes for the code style? 

```suggestion
    directory: `${__dirname}/snippets/`
```